File: auction-frontend/src/app/models/entity/Bid.ts
export class Bid {
  bidId?: string;
  bidderUsername?: string;
  bidPrice?: number;
  bidDate?: string;

  constructor(bidId?: string, bidderUsername?: string, bidPrice?: number, bidDate?: string) {
    this.bidId = bidId;
    this.bidderUsername = bidderUsername;
    this.bidPrice = bidPrice;
    this.bidDate = bidDate;
  }
}

File: auction-backend/src/main/java/com/trkgrn/auctionbackend/api/model/entity/Bid.java
package com.trkgrn.auctionbackend.api.model.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.Column;
import javax.persistence.Embeddable;

@Embeddable
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class Bid {

    @Column(name = "bid_id")
    private String bidId;

    @Column(name = "bid_username")
    private String bidderUsername;

    @Column(name = "bid_price")
    private Double bidPrice;

    @Column(name = "bid_date")
    private String bidDate;

}
